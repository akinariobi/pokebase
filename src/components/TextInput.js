import React from "react";
import { Input } from "antd";

class TextInput extends React.Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.onQueryUpdate(event.target.value);
  }

  render() {
    return (
      <div className="TextInput__container">
        <div style={{ marginBottom: 16 }}>
          <Input
            onChange={this.handleChange}
            placeholder="Name or type of pokÃ©mon"
            defaultValue=""
          />
        </div>
      </div>
    );
  }
}

export default TextInput;
