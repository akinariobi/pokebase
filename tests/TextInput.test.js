import React from "react";
import TextInput from "../src/components/TextInput";
import renderer from "react-test-renderer";

describe("WeaknessFilter", () => {
  test("renders correctly", () => {
    const component = renderer.create(<TextInput />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking if this is a stateless component", () => {
  test("emit events", () => {
    let tree = renderer.create(<TextInput />);

    let instance = tree.getInstance();

    //const event = {
    //  preventDefault() {},
    //  target: { value: 'the-value' }
    //};

    //instance.handleChange(event);

    expect(instance.state).toBe(null);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
