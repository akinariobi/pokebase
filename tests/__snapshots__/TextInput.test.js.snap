// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changing our button name to Hide toggle the button 1`] = `
<div
  className="TextInput__container"
>
  <div
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <input
      className="ant-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Name or type of pokémon"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`WeaknessFilter renders correctly 1`] = `
<div
  className="TextInput__container"
>
  <div
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <input
      className="ant-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Name or type of pokémon"
      type="text"
      value=""
    />
  </div>
</div>
`;
